crates/ruma-signatures/src/keys.rs

```diff
@@ -42,27 +90,59 @@ impl Ed25519KeyPair {
     ///
     /// Returns an error if the public and private keys provided are invalid for the implementing
     /// algorithm.
-    pub fn new(document: &[u8], version: String) -> Result<Self, Error> {
-        let keypair = RingEd25519KeyPair::from_pkcs8(document)
-            .map_err(|error| Error::new(error.to_string()))?;
+    ///
+    /// Returns an error when the PKCS#8 document had a public key, but it doesn't match the one
+    /// generated from the private key. This is a fallback and extra validation against
+    /// corruption or
+    pub fn from_der(document: &[u8], version: String) -> Result<Self, Error> {
```

Comment:
`new` -> `from_der`

This was OK'd by @jplatte on the basis that this branch will later merge into `next`